### DISCUSSION TITLE
Valid Email from Fullname()

### DISCUSSION THREAD

**Discussion by yudyananda** [faker-js/faker#1881] [Q&A]
I just find out that generated `email` from `fullName` contain dot will produce invalid email address (double dot) For example **Dwight Armstrong Jr.**  => **dwightarmstrongjr..kerluke@xxxx**. Is this intended or am I doing it the wrong way?

```typescript
const prisma = new PrismaClient();

async function main() {
  for (let i = 0; i < 30; i++) {
    const fullName = faker.name.fullName();

    const user = await prisma.user.create({
      data: {
        name: fullName,
        email: faker.internet.email(fullName).toLocaleLowerCase(),
        image: '/avatar.png',
      },
    });

    await prisma.pendaftar.create({
      data: {
        user: {
          connect: {
            id: user.id,
          },
        },
      },
    });
  }
}
```

**Comment by matthewmayer** [comment]
If you only pass one parameter to faker.internet.email it treats it as a firstname, and adds a random lastname (in your example "kerluke").

Probably what you want to do instead is

```
const firstName = faker.name.firstName();
const lastName = faker.name.lastName();
const fullName = faker.name.fullName({firstName, lastName});
const email = faker.internet.email(firstName, lastName).toLocaleLowerCase(),

    const user = await prisma.user.create({
      data: {
        name: fullName,
        email: email,
        image: '/avatar.png',
      },
    });
```

**Reply by Shinigami92** [reply]
This answer is just fighting the symptome and not addressing the bug

We have already opened a bug report here
- https://github.com/faker-js/faker/issues/1882

and track it
